server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -152,6 +154,32 @@ public void shardRoutingChanged(IndexShard indexShard, @Nullable ShardRouting ol
         }
     }
 
+    /**
+     * returns SegmentReplicationState of on-going segment replication events.
+     */
+    public SegmentReplicationState getOngoingEventSegmentReplicationState(ShardRouting shardRouting) {
+        SegmentReplicationTarget target = onGoingReplications.getOngoingReplicationTarget(shardRouting.shardId());
```

Comment:
Please add `@Nullable` annotation to these method declarations. Also all three of these methods only need the ShardId, not the entire ShardRouting.

Also nit - 
```java
        return Optional.ofNullable(onGoingReplications.getOngoingReplicationTarget(shardRouting.shardId()))
            .map(SegmentReplicationTarget::state)
            .orElse(null);
```